src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -64,32 +63,35 @@ public boolean addAll(List<T> list) {
     @Override
     public T get(int index) {
         checkIndex(index);
-        return node(index).value;
+        return getNode(index).value;
 
     }
 
     @Override
     public T set(T value, int index) {
         checkIndex(index);
-        T returnedValue = node(index).value;
-        node(index).value = value;
+        T returnedValue = getNode(index).value;
+        getNode(index).value = value;
         return returnedValue;
     }
 
     @Override
     public T remove(int index) {
         checkIndex(index);
-        Node<T> removedValue = node(index);
+        Node<T> removedValue = getNode(index);
         return unlink(removedValue);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
         for (int i = 0; i < size; i++) {
-            if (node(i).value == object || object != null && object.equals(node(i).value)) {
-                remove(i);
+            if (current.value == object || (object != null
+                    && object.equals(current.value))) {
+                unlink(current);
```

Comment:
Here JDK offers to use Object.equals() - it's correct to change to this checking?

Reply:
at the moment - we should not use `Object.equals()`
we should learn to write it by hands